examples/skinned_meshlets/source/skinned_meshlets.cpp

```diff
@@ -0,0 +1,628 @@
+			gvk::context().main_window()->set_cursor_pos({ resolution[0] / 2.0, resolution[1] / 2.0 });
+		}
+		if (gvk::input().key_pressed(gvk::key_code::escape)) {
+			// Stop the current composition:
+			gvk::current_composition()->stop();
+		}
+		if (gvk::input().key_pressed(gvk::key_code::left)) {
+			mQuakeCam.look_along(gvk::left());
+		}
+		if (gvk::input().key_pressed(gvk::key_code::right)) {
+			mQuakeCam.look_along(gvk::right());
+		}
+		if (gvk::input().key_pressed(gvk::key_code::up)) {
+			mQuakeCam.look_along(gvk::front());
+		}
+		if (gvk::input().key_pressed(gvk::key_code::down)) {
+			mQuakeCam.look_along(gvk::back());
+		}
+		if (gvk::input().key_pressed(gvk::key_code::page_up)) {
+			mQuakeCam.look_along(gvk::up());
+		}
+		if (gvk::input().key_pressed(gvk::key_code::page_down)) {
+			mQuakeCam.look_along(gvk::down());
+		}
+		if (gvk::input().key_pressed(gvk::key_code::home)) {
+			mQuakeCam.look_at(glm::vec3{ 0.0f, 0.0f, 0.0f });
+		}
+
+		if (gvk::input().key_pressed(gvk::key_code::f1)) {
+			auto imguiManager = gvk::current_composition()->element_by_type<gvk::imgui_manager>();
+			if (mQuakeCam.is_enabled()) {
+				mQuakeCam.disable();
+				if (nullptr != imguiManager) { imguiManager->enable_user_interaction(true); }
+			}
+			else {
+				mQuakeCam.enable();
+				if (nullptr != imguiManager) { imguiManager->enable_user_interaction(false); }
+			}
+		}
+
+		// Animate all the meshes
+		for (auto& model : mAnimatedModels) {
+			auto& animation = std::get<animated_model_data>(model).mAnimation;
+			auto& clip = std::get<animated_model_data>(model).mClip;
+			const auto doubleTime = fmod(gvk::time().absolute_time_dp(), std::get<animated_model_data>(model).duration_sec() * 2);
+			auto time = glm::lerp(std::get<animated_model_data>(model).start_sec(), std::get<animated_model_data>(model).end_sec(), (doubleTime > std::get<animated_model_data>(model).duration_sec() ? doubleTime - std::get<animated_model_data>(model).duration_sec() : doubleTime) / std::get<animated_model_data>(model).duration_sec());
+			auto targetMemory = std::get<additional_animated_model_data>(model).mBoneMatricesAni.data();
+
+			// Use lambda option 1 that takes as parameters: mesh_bone_info, inverse mesh root matrix, global node/bone transform w.r.t. the animation, inverse bind-pose matrix
+			animation.animate(clip, time, [this, &animation, targetMemory](gvk::mesh_bone_info aInfo, const glm::mat4& aInverseMeshRootMatrix, const glm::mat4& aTransformMatrix, const glm::mat4& aInverseBindPoseMatrix, const glm::mat4& aLocalTransformMatrix, size_t aAnimatedNodeIndex, size_t aBoneMeshTargetIndex, double aAnimationTimeInTicks) {
+				// Construction of the bone matrix for this node:
+				//   1. Bring vertex into bone space
+				//   2. Apply transformaton in bone space => MODEL SPACE
+				targetMemory[aInfo.mGlobalBoneIndexOffset + aInfo.mMeshLocalBoneIndex] = aTransformMatrix * aInverseBindPoseMatrix;
+			});
+		}
```

Comment:
I think it would be better to have this code in `render()`, because actually, this code is only related to the rendering of the scene. There is a component (`fixed_update_timer`), which has `update()` and `render()` invoked _not_ in sync. It does actually not make sense to update bone animation matrices more or less often than rendering them. On the contrary, they shall _always_ be updated right before rendering. I.e., just move this code to the beginning of `render()`. Or, even better, put it in a member function (for better code structuring) and invoke that one!